python/tvm/_ffi/runtime_ctypes.py

```diff
@@ -197,6 +199,10 @@ def _GetDeviceAttr(self, device_type, device_id, attr_id):
     @property
     def exist(self):
         """Whether this device exist."""
+        allowed_ctxs = os.environ.get("TVM_TEST_CTXS")
```

Comment:
This seems to be a hack that we should not put in here, but instead put in tvm.testing

Reply:
Sorry, this is old code. Forgot to remove it.